clients/src/main/java/org/apache/kafka/common/serialization/Serde.java

```diff
@@ -13,12 +13,32 @@
 
 package org.apache.kafka.common.serialization;
 
+import java.io.Closeable;
+import java.util.Map;
+
 /**
  * The interface for wrapping a serializer and deserializer for the given data type.
  *
- * @param <T>
+ * @param <T> Type to be serialized from and deserialized into.
+ *
+ * A class that implements this interface is expected to have a constructor with no parameter.
  */
-public interface Serde<T> {
+public interface Serde<T> extends Closeable {
+
+    /**
+     * Configure this class, which will configure the underlying serializer and deserializer at the same time.
+     *
+     * @param configs configs in key/value pairs
+     * @param isKey whether is for key or value
+     */
+    void configure(Map<String, ?> configs, boolean isKey);
+
+    /**
+     * Close this serde class, which will close the underlying serializer and deserializer at the same time.
+     * This method has to be idempotent because it might be called multiple times.
+     */
+    @Override
+    void close();
```

Comment:
This is a nitpick, but perhaps we should not use `at the same time` in `close` or `configure` because it won't actually happen at the same time (it will most likely happen sequentially, but we don't have to specify that).